### ISSUE TITLE
Devel branch should have higher version than master

### ISSUE THREAD

**Issue by andre-merzky** [radical-cybertools/radical.pilot#2815]
TODO AM: automate

**Comment by andre-merzky** [comment]
I adapted the release procedure to implement this proposal.  Now, the release procedure is basically two scripts which I run locally.  I'll add them here for documentation.  We may want to automate this at some point - but I personally like the flexibility we have with the current release process, crafting the changelog manually (which removes lot of noise), etc.

preparing a release:
```sh
$ git prep
$ git release
$ git bump minor
```

`git prep` is a git script for the following steps
```sh
$ git co master --> checkout master
$ git pa  --> pull all branches form remote
$ git gone -a --> remove all stale branches
$ git merge devel --> merge the release candidate
$ git change >> CHANGES.md --> create a draft changelog from the set of new commits in master
$ gvim -o VERSION CHANGES.md --> set new version number (if not yet set, see next part), make CHANGELOG human readable  
```

After that that last manual intervention, the actual release itself with `git release` runs::
```sh
$ git ci -am 'version bump' --> commit VERSION and CHANGELOG changes
$ git tag \"v$(cat VERSION)\" --> tag the release
$ git push --> push master to origin
$ git push --tags --> push the release tag
$ make upload  --> push to pypi
$ git co devel --> checkout devel
$ git merge master --> merge the release into devel
$ git bump minor --> bump minor version
$ git ci -am 'devel version bump' --> commit minor version bump on devel
$ git dync -a --> sync all branches to be in sync with devel
```

That last step is hard to automate as it involves resolving conflicts in all branches.  But it is also important - it saved us over the last years from branches running out of sync with devel.  `git-dsync` is a custom script - happy to share if anybody is interested.

Long story short: the proposed change is now implemented and will be triggered on the next release.
